network/transport/v2/scheduler.go

```diff
@@ -173,7 +191,8 @@ func (p *payloadScheduler) retry(hash hash.SHA256Hash, initialCount uint16) {
 			}),
 		)
 		if err != nil {
-			log.Logger().Errorf("Failed to pass payload query to network: %v", err)
+			log.Logger().Errorf("failed to pass payload query to network: %v", err)
+			p.addFailedJob(hash)
```

Comment:
job is never removed? Is this the list of TXs that ultimately failed after many retries? Or TXs that are still being retried? If the first, the second is useful too, I think.

If they're retried no more, does the diagnostic survive restarts (since it's in-memory)?

Reply:
> job is never removed? Is this the list of TXs that ultimately failed after many retries? Or TXs that are still being retried? If the first, the second is useful too, I think.

It's the list of txs that failed after 100 retries.

> If the first, the second is useful too, I think.

Could be true but wouldn't want to do that right now as it's specifically for debugging private transactions that were never fully processed.

> If they're retried no more, does the diagnostic survive restarts (since it's in-memory)?
 Write

I thought about this as well but it shouldn't matter as the retry DAG publisher should also retry these jobs right?